source/fuzz/fuzzer_pass_add_loads.cpp

```diff
@@ -79,13 +83,36 @@ void FuzzerPassAddLoads::Apply() {
           return;
         }
 
+        auto choosen_instruction =
+            relevant_instructions[GetFuzzerContext()->RandomIndex(
+                relevant_instructions)];
+
+        bool is_atomic_load = false;
+        uint32_t memory_scope_id = 0;
+        uint32_t memory_semtantics_id = 0;
+
+        // has work group or storage buffer (storage class)
+        if (choosen_instruction) {
+          if (GetFuzzerContext()->ChoosePercentage(
+                  GetFuzzerContext()->GetChanceOfAddingAtomicLoad())) {
+            is_atomic_load = true;
+            memory_scope_id =
+                FindOrCreateConstant({SpvScopeInvocation},
+                                     FindOrCreateIntegerType(32, false), false);
+            memory_semtantics_id =
+                FindOrCreateConstant(/*NOTE: I think the value(Words) here needs
+                                        to be selected under a condition? */
+                                     {SpvMemorySemanticsWorkgroupMemoryMask,
+                                      SpvMemorySemanticsUniformMemoryMask},
```

Comment:
You are creating a 32 bit integer with 2 words (i.e. 64 bits). This is incorrect. Use `SpvMemorySemanticsWorkgroupMemoryMask` or `SpvMemorySemanticsUniformMemoryMask`, but not both. The choice depends on whether choosen_instruction points at workgroup memory or storage buffer memory.